MacDown/Code/Utility/MPHomebrewSubprocessController.m

```diff
@@ -0,0 +1,91 @@
+//
+//  MPHomebrewSubprocessController.m
+//  MacDown
+//
+//  Created by Tzu-ping Chung on 18/2.
+//  Copyright © 2017 Tzu-ping Chung . All rights reserved.
+//
+
+#import "MPHomebrewSubprocessController.h"
+
+
+@interface MPHomebrewSubprocessController ()
+
+@property (readonly) NSTask *task;
+@property (readonly) void(^completionHandler)(NSString *);
+
+@end
+
+
+@implementation MPHomebrewSubprocessController
+
+- (instancetype)initWithArguments:(NSArray *)args
+{
+    self = [super init];
+    if (!self)
+        return nil;
+
+    NSPipe *stdoutPipe = [[NSPipe alloc] init];
+    NSFileHandle *stdoutReadHandle = stdoutPipe.fileHandleForReading;
+
+    _task = [[NSTask alloc] init];
+    _task.launchPath = @"brew";
+    if (args)
+        _task.arguments = args;
+    _task.standardOutput = stdoutPipe;
+
+    NSNotificationCenter *center = [NSNotificationCenter defaultCenter];
+    [center addObserver:self
+               selector:@selector(homebrewReadDidComplete:)
+                   name:NSFileHandleReadToEndOfFileCompletionNotification
+                 object:stdoutReadHandle];
+    [stdoutReadHandle readToEndOfFileInBackgroundAndNotify];
+
+    return self;
+}
+
+- (instancetype)init
+{
+    return [self initWithArguments:nil];
+}
+
+- (void)runWithCompletionHandler:(void(^)(NSString *))handler
+{
+    _completionHandler = handler;
+    @try {
```

Comment:
Should this { be on a separate line? Does { on new line for `if`, `for` etc from the MacDown coding style(_CONTRIBUTING.md_) also apply to @try and @catch?